pkg/node/node.go

```diff
@@ -696,24 +696,49 @@ func NewBee(interrupt chan os.Signal, addr string, publicKey *ecdsa.PublicKey, s
 	batchStore.SetRadiusSetter(kad)
 
 	if batchSvc != nil && chainEnabled {
-		syncedChan, err := batchSvc.Start(postageSyncStart, initBatchState)
-		if err != nil {
-			return nil, fmt.Errorf("unable to start batch service: %w", err)
-		}
-		// wait for the postage contract listener to sync
-		logger.Info("waiting to sync postage contract data, this may take a while... more info available in Debug loglevel")
+		if o.FullNodeMode {
+			syncedChan, err := batchSvc.Start(postageSyncStart, initBatchState)
 
-		// arguably this is not a very nice solution since we dont support
-		// interrupts at this stage of the application lifecycle. some changes
-		// would be needed on the cmd level to support context cancellation at
-		// this stage
-		select {
-		case err = <-syncedChan:
 			if err != nil {
-				return nil, err
+				batchSvc.Set(err)
+				return nil, fmt.Errorf("unable to start batch service: %w", err)
+			}
+			// wait for the postage contract listener to sync
+			logger.Info("waiting to sync postage contract data, this may take a while... more info available in Debug loglevel")
+
+			// arguably this is not a very nice solution since we dont support
+			// interrupts at this stage of the application lifecycle. some changes
+			// would be needed on the cmd level to support context cancellation at
+			// this stage
+			select {
+			case err = <-syncedChan:
+				batchSvc.Set(err)
+				if err != nil {
+					return nil, err
+				}
+			case <-interrupt:
+				return nil, ErrInterruped
 			}
-		case <-interrupt:
-			return nil, ErrInterruped
+		} else {
+			go func() {
+				logger.Info("started postage contract data sync in the background...")
+				syncedChan, err := batchSvc.Start(postageSyncStart, initBatchState)
```

Comment:
this is almost identical to the code block above. i'd refactor this into a local, function scoped function (i.e. `syncFn := func() { ....}`) and just call it either from the goroutine or from the main thread.